sync/validation_queue.go

```diff
@@ -64,33 +64,37 @@ func (vq *blockQueue) inQueue(id types.Hash32) bool {
 
 // handles all fetched blocks
 // this handler is passed to fetchQueue which is responsible for triggering the call
-func (vq *blockQueue) handleBlock(bjb fetchJob) {
+func (vq *blockQueue) handleBlocks(bjb fetchJob) {
 	mp := map[types.Hash32]*types.Block{}
 	for _, item := range bjb.items {
 		tmp := item.(*types.Block)
 		mp[item.Hash32()] = tmp
 	}
 
 	for _, id := range bjb.ids {
-
-		block, found := mp[id]
-		if !found {
+		b, ok := mp[id]
+		if !ok {
 			vq.updateDependencies(id, false)
 			vq.Error("could not retrieve a block in view %v", id.ShortString())
 			continue
 		}
 
+		go vq.handleBlock(id, b)
+	}
+
+}
+
+func (vq *blockQueue) handleBlock(id types.Hash32, b *types.Block) {
+	func(block *types.Block) {
```

Comment:
why call this inner function? you just create a wrapping function and call it...

Reply:
auto refactor, will fiz